dev/publish_bom.py

```diff
@@ -0,0 +1,193 @@
+      gist = self.__github.get_user().create_gist(True, {filename: content}, description=description)
+      print ('Wrote changelog to Gist at https://gist.github.com/{user}/{id}'
+             .format(user=self.__gist_user, id=gist.id))
+
+  def push_branch_and_tags(self):
+    """Creates a release branch and pushes tags to the microservice repos owned by --github_publisher.
+
+    A private key that has access to --github_publisher's github repos needs added
+    to a running ssh-agent on the machine this script will run on:
+
+    > <copy or rsync the key to the vm>
+    > eval `ssh-agent`
+    > ssh-add ~/.ssh/<key with access to github repos>
+
+    """
+    major, minor, _ = self.__release_version.split('.')
+
+    # The stable branch will look like <major>.<minor>.X since nebula
+    # enforces restrictions on what branches it does releases from.
+    # https://github.com/nebula-plugins/nebula-release-plugin#extension-provided
+    stable_branch = '.'.join([major, minor, 'X'])
+    for comp in COMPONENTS:
+      if self.__patch_release:
+        self.__exit_if_failure('git -C {0} checkout {1}'.format(comp, stable_branch),
+                               'git checkout')
+      else:
+        # Create new release branch.
+        self.__exit_if_failure('git -C {0} checkout -b {1}'.format(comp, stable_branch),
+                               'git checkout')
+
+      version_tag_build = ''
+      if comp == 'spinnaker-monitoring':
+        version_tag_build = 'version-{0}'.format(self.__bom_dict[SERVICES]['monitoring-daemon'][VERSION])
+      else:
+        version_tag_build = 'version-{0}'.format(self.__bom_dict[SERVICES][comp][VERSION])
+
+      last_dash = version_tag_build.rindex('-')
+      version_tag = version_tag_build[:last_dash]
+      repo_to_push = ('git@github.com:{owner}/{comp}.git'
+                      .format(owner=self.__github_publisher, comp=comp))
+      self.__exit_if_failure(
+        'git -C {comp} remote add release {url}'.format(comp=comp, url=repo_to_push),
+        'git remote add'
+      )
+      self.__exit_if_failure(
+        'git -C {comp} push release {branch}'.format(comp=comp,  branch=stable_branch),
+        'git push branch'
+      )
+
+      repo = self.__github.get_repo('{owner}/{comp}'.format(owner=self.__github_publisher, comp=comp))
+      paginated_tags = repo.get_tags()
+      tag_names = [tag.name for tag in paginated_tags]
+      if version_tag not in tag_names:
+        # The tag doesn't exist and we need to push a tag.
+        print ('pushing version tag {tag} to {owner}/{comp}'
+               .format(tag=version_tag, owner=self.__github_publisher, comp=comp))
+        self.__exit_if_failure(
+          'git -C {comp} push release {tag}'.format(comp=comp,  tag=version_tag),
+          'git push tag'
+        )
+
+  def __exit_if_failure(self, command, task_name):
+    """Helper function to run_quick a command and exit if it fails.
+    """
+    result = run_quick(command)
```

Comment:
this is check_run_quick (which raises RuntimeError but you can trap that top level)